Swap `jsonwebtoken` for `jose` to offer edge compatibility

The `verify` method of the `jsonwebtoken` package [depends on the Node.js `crypto` module](https://github.com/auth0/node-jsonwebtoken/blob/bc28861f1fa981ed9c009e29c044a19760a0b128/verify.js#L9). This means that calling [`verify`](https://github.com/matthewvolk/bigrequest/blob/0e50eeb68cf9e140ebbe83b6b4e28a7baa148e5c/packages/bigrequest/src/oauth.ts#L86-L135) in an edge runtime that does not have access to all Node.js APIs (for example, [Vercel's edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes), [Cloudflare Workers](https://developers.cloudflare.com/workers/runtime-apis/), [Deno](https://deno.land/api), [Bun](https://bun.sh/docs/runtime/web-apis), etc.) results in the following error:

```
Error: Module not found: Can't resolve 'crypto'

Import trace for requested module:
./node_modules/.pnpm/jsonwebtoken@9.0.2/node_modules/jsonwebtoken/index.js
./node_modules/.pnpm/bigrequest@0.0.15/node_modules/bigrequest/dist/index.mjs
```

We should swap the `jsonwebtoken` module with an edge-compatible module that can verify JWT's, such as [jose](https://www.npmjs.com/package/jose), so that `verify` can be called in both Node.js and Edge runtimes.

### Acceptance Criteria 
The `verify` method of the `bigrequest.oauth` module can be called when executing in the following runtimes:
* [Vercel's edge runtime](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes)
* [Cloudflare Workers](https://developers.cloudflare.com/workers/runtime-apis/)
* [Deno](https://deno.land/api)
* [Bun](https://bun.sh/docs/runtime/web-apis)

--- a/.changeset/weak-eggs-work.md
+++ b/.changeset/weak-eggs-work.md
@@ -1,0 +1,6 @@
+---
+"bigrequest": patch
+"bigexec": patch
+---
+
+Closes #123 - Swaps `jsonwebtoken` for `jose` in order to offer edge-compatibility when using the `oauth.verify` method.

--- a/packages/bigrequest/package.json
+++ b/packages/bigrequest/package.json
@@ -27,13 +27,12 @@
     "test": "echo \"No test specified\""
   },
   "dependencies": {
-    "jsonwebtoken": "^9.0.1",
+    "jose": "^5.1.3",
     "openapi-fetch": "^0.8.1",
     "zod": "^3.21.4"
   },
   "devDependencies": {
     "@bigcommerce/eslint-config": "^2.7.0",
-    "@types/jsonwebtoken": "^9.0.2",
     "@types/node": "18.0.0",
     "eslint": "^8.46.0",
     "openapi-typescript": "^6.4.0",

--- a/packages/bigrequest/src/oauth.ts
+++ b/packages/bigrequest/src/oauth.ts
@@ -1,4 +1,4 @@
-import { verify as verifyJwt } from 'jsonwebtoken';
+import { jwtVerify } from 'jose';
 import { z } from 'zod';
 
 import { BigRequestError } from './error';
@@ -83,7 +83,7 @@ export const oauth = (config: z.infer<typeof oauthConfigSchema>) => {
 
   const jwtSchema = z.string().min(1);
 
-  const verify = (signedPayloadJwt: z.infer<typeof jwtSchema>) => {
+  const verify = async (signedPayloadJwt: z.infer<typeof jwtSchema>) => {
     const parsedJwt = jwtSchema.safeParse(signedPayloadJwt);
 
     if (!parsedJwt.success) {
@@ -96,7 +96,9 @@ export const oauth = (config: z.infer<typeof oauthConfigSchema>) => {
       );
     }
 
-    const decoded = verifyJwt(parsedJwt.data, oauthConfig.data.clientSecret);
+    const secret = new TextEncoder().encode(oauthConfig.data.clientSecret);
+
+    const decoded = await jwtVerify(parsedJwt.data, secret);
 
     const verifiedJwtSchema = z.object({
       aud: z.string(),
@@ -119,7 +121,7 @@ export const oauth = (config: z.infer<typeof oauthConfigSchema>) => {
       channel_id: z.number().nullable(),
     });
 
-    const parsedVerifiedJwt = verifiedJwtSchema.safeParse(decoded);
+    const parsedVerifiedJwt = verifiedJwtSchema.safeParse(decoded.payload);
 
     if (!parsedVerifiedJwt.success) {
       throw new BigRequestError(

